### ISSUE TITLE
Clean C++20 support for CMake 3.12+

### ISSUE THREAD

**Issue by Lyberta** [conan-io/conan#3327]
CMake 3.12 is out there and officially supports C++20. So now you can use clean way to enable C++20 mode if CMake 3.12 or later is detected.

**Comment by Croydon** [comment]
What are you asking for with this issue?

**Comment by lasote** [comment]
I think Conan already does correctly what you are requesting:

```
macro(conan_set_std)
  # Do not warn "Manually-specified variables were not used by the project"
  set(ignorevar "${CONAN_STD_CXX_FLAG}${CONAN_CMAKE_CXX_STANDARD}${CONAN_CMAKE_CXX_EXTENSIONS}")
  # TODO: Check version where CMake supports 20, still not released
  if (CMAKE_VERSION VERSION_LESS "3.1" OR "${CONAN_CMAKE_CXX_STANDARD}" STREQUAL "20"
                                       OR "${CONAN_CMAKE_CXX_STANDARD}" STREQUAL "gnu20")
    if(CONAN_STD_CXX_FLAG)
      message(STATUS "Conan setting CXX_FLAGS flags: ${CONAN_STD_CXX_FLAG}")
      set(CMAKE_CXX_FLAGS "${CONAN_STD_CXX_FLAG} ${CMAKE_CXX_FLAGS}")
    endif()
  else()
    if(CONAN_CMAKE_CXX_STANDARD)
      message(STATUS "Conan setting CPP STANDARD: ${CONAN_CMAKE_CXX_STANDARD} WITH EXTENSIONS ${CONAN_CMAKE_CXX_EXTENSIONS}")
      set(CMAKE_CXX_STANDARD ${CONAN_CMAKE_CXX_STANDARD})
      set(CMAKE_CXX_EXTENSIONS ${CONAN_CMAKE_CXX_EXTENSIONS})
    endif()
  endif ()
endmacro()
```

**Comment by Lyberta** [comment]
The answer is in the code you've posted:

    # TODO: Check version where CMake supports 20, still not released

**Comment by lasote** [comment]
You are totally right. I'll open a PR asap.

**Comment by lasote** [comment]
Will be released in 0.17. Many thanks for your help!
